src/main/java/com/sapred/ordermanagerred/service/OrderService.java

```diff
@@ -54,6 +64,34 @@ public List<Order> getOrders(String token, String statusId, int pageNumber, Stri
         return pageOrders.getContent();
     }
 
+    public List<Order> getOrdersByFilters(Map<String, Object> filterMap, String token, int pageNumber) {
+        if (!filterMap.containsKey(Order.Fields.companyId)) {
+            throw new MapFilterMissedField("the companyId key is require!");
+        }
+
+        Criteria criteria = new Criteria();
+
+        // Iterate through the filter map and construct filter for each entry
+        for (Map.Entry<String, Object> entry : filterMap.entrySet()) {
+            String filterName = entry.getKey();
+            Object filterValue = entry.getValue();
+
+            criteria = criteria.and(filterName).is(filterValue);
+        }
+
+        Query query = new Query(criteria);
+
+        int skip = (pageNumber - 1) * pageSize;
```

Comment:
Why did you change from filter to criteria?

Reply:
because this way: orderCollection.find(finalFilter, Order.class)  worked for me just like mongotemplate.find(...) and this got criteria.